temporian/core/data/duration.py

```diff
@@ -36,3 +41,65 @@ def hours(value: float) -> Duration:
 
 def days(value: float) -> Duration:
     return value * 60 * 60 * 24
+
+
+supported_date_types = (np.datetime64, time.struct_time, datetime.datetime)
+
+
+def convert_date_to_duration(
+    date: Union[np.datetime64, time.struct_time, datetime.datetime]
+) -> Duration:
+    """Convert date to duration epoch UTC
+
+    Args:
+        date (Union[np.datetime64, time.struct_time, datetime.datetime]):
+            Date to convert
+
+    Returns:
+        int: Duration epoch UTC
+
+    Raises:
+        TypeError: Unsupported type. Supported types are:
+        - np.datetime64
+        - time.struct_time
+        - datetime.datetime
+    """
+    if isinstance(date, np.datetime64):
+        return convert_numpy_datetime64_to_duration(date)
+    if isinstance(date, time.struct_time):
+        return convert_struct_time_to_duration(date)
+    if isinstance(date, datetime.datetime):
+        return convert_datetime_to_duration(date)
+
+    raise TypeError(f"Unsupported type: {type(date)}")
+
+
+def convert_numpy_datetime64_to_duration(date: np.datetime64) -> Duration:
+    """Convert numpy datetime64 to duration epoch UTC"""
+    return date.astype("datetime64[s]").astype("float64")
+
+
+def convert_struct_time_to_duration(date: time.struct_time) -> Duration:
+    """Convert struct_time to duration epoch UTC"""
+    return float(time.mktime(date) - time.timezone)
+
+
+def convert_datetime_to_duration(date: datetime.datetime) -> Duration:
+    """Convert datetime to duration epoch UTC"""
+    date = date.replace(tzinfo=datetime.timezone.utc)
+    return float(date.timestamp())

```

Comment:
If it was a float, it would return false in the `is_a_date` function and therefore no conversion will be needed.  This only converts datetime objects.